internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -54,30 +67,234 @@ var _ = Describe("VM Console Proxy Operand", func() {
+		// ExpectResourceNotExists(bundle.configMap, request)
+		// ExpectResourceNotExists(bundle.service, request)
+		// ExpectResourceNotExists(bundle.deployment, request)
+		ExpectResourceNotExists(newRoute(namespace, "vm-console-proxy"), request)
+	})
+
+	It("should not update service cluster IP", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		key := client.ObjectKeyFromObject(bundle.service)
+		service := &core.Service{}
+		Expect(request.Client.Get(request.Context, key, service)).ToNot(HaveOccurred())
+
+		// This address is from a range of IP addresses reserved for documentation.
+		const testClusterIp = "198.51.100.42"
+
+		service.Spec.ClusterIP = testClusterIp
+		Expect(request.Client.Update(request.Context, service)).ToNot(HaveOccurred())
+
+		_, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		updatedService := &core.Service{}
+		Expect(request.Client.Get(request.Context, key, updatedService)).ToNot(HaveOccurred())
+		Expect(updatedService.Spec.ClusterIP).To(Equal(testClusterIp))
+	})
+
+	It("should report status", func() {
+		reconcileResults, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Set status for deployment
+		key := client.ObjectKeyFromObject(bundle.deployment)
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = 0
+			deployment.Status.AvailableReplicas = 0
+			deployment.Status.UpdatedReplicas = 0
+			deployment.Status.UnavailableReplicas = replicas
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Only deployment should be progressing
+		for _, reconcileResult := range reconcileResults {
+			if _, ok := reconcileResult.Resource.(*apps.Deployment); ok {
+				Expect(reconcileResult.Status.NotAvailable).ToNot(BeNil())
+				Expect(reconcileResult.Status.Progressing).ToNot(BeNil())
+				Expect(reconcileResult.Status.Degraded).ToNot(BeNil())
+			} else {
+				Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+				Expect(reconcileResult.Status.Progressing).To(BeNil())
+				Expect(reconcileResult.Status.Degraded).To(BeNil())
+			}
+		}
+
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = replicas
+			deployment.Status.AvailableReplicas = replicas
+			deployment.Status.UpdatedReplicas = replicas
+			deployment.Status.UnavailableReplicas = 0
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// All resources should be available
+		for _, reconcileResult := range reconcileResults {
+			Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+			Expect(reconcileResult.Status.Progressing).To(BeNil())
+			Expect(reconcileResult.Status.Degraded).To(BeNil())
+		}
+	})
+
+	Describe("GetVmConsoleProxyImage", func() {
```

Comment:
Unit tests should not test private functions. Can you rewrite this in terms of the created deployment?
Or you can just remove it.